examples/testgputext.c

```diff
@@ -0,0 +1,401 @@
+            .offset = 0 },
+        &(SDL_GPUBufferRegion){
+            .buffer = context->vertex_buffer,
+            .offset = 0,
+            .size = sizeof(Vertex) * geometry_data->vertex_count },
+        false);
+    SDL_UploadToGPUBuffer(
+        copy_pass,
+        &(SDL_GPUTransferBufferLocation){
+            .transfer_buffer = context->transfer_buffer,
+            .offset = sizeof(Vertex) * MAX_VERTEX_COUNT },
+        &(SDL_GPUBufferRegion){
+            .buffer = context->index_buffer,
+            .offset = 0,
+            .size = sizeof(int) * geometry_data->index_count },
+        false);
+    SDL_EndGPUCopyPass(copy_pass);
+}
+
+void draw(Context *context, SDL_Mat4X4 *matrices, int num_matrices, TTF_GPUAtlasDrawSequence *draw_sequence)
+{
+    SDL_GPUTexture *swapchain_texture;
+    check_error_bool(SDL_AcquireGPUSwapchainTexture(context->cmd_buf, context->window, &swapchain_texture, NULL, NULL));
+
+    if (swapchain_texture != NULL) {
+        SDL_GPUColorTargetInfo colour_target_info = { 0 };
+        colour_target_info.texture = swapchain_texture;
+        colour_target_info.clear_color = (SDL_FColor){ 0.3f, 0.4f, 0.5f, 1.0f };
+        colour_target_info.load_op = SDL_GPU_LOADOP_CLEAR;
+        colour_target_info.store_op = SDL_GPU_STOREOP_STORE;
+
+        SDL_GPURenderPass *render_pass = SDL_BeginGPURenderPass(context->cmd_buf, &colour_target_info, 1, NULL);
+
+        SDL_BindGPUGraphicsPipeline(render_pass, context->pipeline);
+        SDL_BindGPUVertexBuffers(
+            render_pass, 0,
+            &(SDL_GPUBufferBinding){
+                .buffer = context->vertex_buffer, .offset = 0 },
+            1);
+        SDL_BindGPUIndexBuffer(
+            render_pass,
+            &(SDL_GPUBufferBinding){
+                .buffer = context->index_buffer, .offset = 0 },
+            SDL_GPU_INDEXELEMENTSIZE_32BIT);
+        SDL_PushGPUVertexUniformData(context->cmd_buf, 0, matrices, sizeof(SDL_Mat4X4) * num_matrices);
+
+        int index_offset = 0;
+        for (TTF_GPUAtlasDrawSequence *seq = draw_sequence; seq != NULL; seq = seq->next) {
+            SDL_BindGPUFragmentSamplers(
+                render_pass, 0,
+                &(SDL_GPUTextureSamplerBinding){
+                    .texture = seq->atlas_texture, .sampler = context->sampler },
+                1);
+
+            SDL_DrawGPUIndexedPrimitives(render_pass, seq->num_indices, 1, index_offset, 0, 0);
+
+            index_offset += seq->num_indices;
+        }
+        SDL_EndGPURenderPass(render_pass);
+    }
+}
+
+void free_context(Context *context)
+{
+    SDL_ReleaseGPUTransferBuffer(context->device, context->transfer_buffer);
+    SDL_ReleaseGPUSampler(context->device, context->sampler);
+    SDL_ReleaseGPUBuffer(context->device, context->vertex_buffer);
+    SDL_ReleaseGPUBuffer(context->device, context->index_buffer);
+    SDL_ReleaseGPUGraphicsPipeline(context->device, context->pipeline);
+    SDL_DestroyGPUDevice(context->device);
```

Comment:
This is missing `SDL_ReleaseWindowFromGPUDevice(context->device, context->window);`